File: Object-Oriented Programming/Java/Exercises/List/src/Ex4.java

public class Ex4 {

	public static void main(String[] args) throws Exception {
		System.out.printf("Digite um inteiro: ");
		int num = EntradaTeclado.leInt();
		// variavel space para controlar quantos 
		//espacos em branco tera em cada linha e
		// tmp como variavel temporaria de controle
		int space = 0, tmp = 0; 
		
		for(int i = num; i > 0; i--) {
			for(int j = i; j > 0; j--) {
				// se houver espacos em branco
				// para serem colocados
				if (space != 0) { 
					// enquanto tmp nao se igualar a 
					// qtd de espacos necessarios
					while (tmp < space) {
						System.out.printf(" ");
						tmp++;
					}
				}
				System.out.printf("*");
			}
			space++;
			tmp = 0;
			System.out.printf("\n");
		}
	}
}

File: Object-Oriented Programming/Python/Exercises/Ex4.py
print("Digite um numero inteiro: ")
num = input()

space = tmp = 0

for i in range(int(num), 0, -1):
    for j in range(i, 0, -1):
        if(space != 0):
            while(tmp < space) :
                print(" ", end="")
                tmp += 1
        print("*", end="")
    space += 1
    tmp = 0
    print("\n")
    